config/peerpods/podvm/parse_hdr.py

```diff
@@ -0,0 +1,160 @@
+        header_size = 8 + 4 + 4
+        # pv_hdr_head size 416
+        pv_hdr_head_size = 8 + 4 + 4 + 12 + 4 + 8 + 8 + 8 + 8 + 160 + 64 + 64 + 64
+
+        after_key_slot_size = 144
+        # pv_hdr_key_slot digest_key + wrapped_key = phkh
+        phkh_size = 32
+
+        hdr_data = f.read(header_size)
+
+        # Unpack the header fields
+        fields = struct.unpack('8sII', hdr_data)
+
+        magic, version, phs = fields
+        # The last 16 bits is the image tag
+        f.seek(-16, 2)
+        image_tag = f.read(16)
+        # Print the extracted fields
+
+        print(f"Magic: {magic.decode('ascii')}")
+        print(f"phs: {phs}")
+
+        f.seek(pv_hdr_head_size)
+
+        length_phkh_data = phs - pv_hdr_head_size - after_key_slot_size
+        phkh_data = f.read(length_phkh_data)
+
+        # Define the struct format (32 bytes for digest_key, 32 bytes for wrapped_key, 16 bytes for tag)
+        struct_format = '32s32s16s'
+
+        # Calculate the size of each struct
+        struct_size = struct.calcsize(struct_format)
+
+        for i in range(0, len(phkh_data), struct_size):
+            if i + struct_size > len(phkh_data):
+                break
+            chunk = phkh_data[i:i + struct_size]
+            digest_key, wrapped_key, tag = struct.unpack(struct_format, chunk)
+            if digest_key.hex() == hkd_phkh:
+                 key_slot_used_idx = i
+                 print(f" ========Host Key Document Hash used in this slot========= ")
+            print(f"  Key Slot: {i//80 + 1}:")
+            print(f"  image_phkh: {digest_key.hex()}")
+            print(f"  wrapped_key: {wrapped_key.hex()}")
+            print(f"  tag: {tag.hex()}")
+        # if the 1 slot selected, the idx is 0
+        if key_slot_used_idx > -1:
+            chunk_used = phkh_data[key_slot_used_idx:key_slot_used_idx + struct_size]
+            digest_key, wrapped_key, tag = struct.unpack(struct_format, chunk_used)
+            print(f" ========Host Key Document Hash used in this slot========= ")
+            print(f"  Key Slot: {key_slot_used_idx//80 + 1}:")
+            print(f"  wrapped_key: {wrapped_key.hex()}")
+            print(f" HKD tag: {tag.hex()}")
+            print(f"  Copy below value and set in rvps ")
+            print(f"  ================================================ ")
+            print(f"  se.image_phkh: {digest_key.hex()}")
+        else:
+            print(f" The HKD file not included when build the SE image ")
+    
+
+        print(f"  se.version: {version}")
+        print(f"  se.tag: {image_tag.hex()}")
+        print(f"  se.attestation_phkh: {hkd_phkh}")
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print(f"Usage: {sys.argv[0]} <SE image header file: hdr.bin   Host Key Document file: HKD.crt >")
+        sys.exit(1)
+
+    hdr_file = sys.argv[1]
+    hkd_file = sys.argv[2]
+    parse_hdr(hdr_file, hkd_file)
```

Comment:
newline issue to be fixed.

Reply:
The file has been removed and I am using the file from opensource repo.